In [1]:
import sqlite3
import os
import csv
os.chdir("E:/EBAC/Material/M41-ImportacionSQLconPython")


In [2]:
conn = sqlite3.connect("RH.db")
cur = conn.cursor()

In [3]:
with open('recursos_humanos.csv') as f:
    reader = csv.reader(f)
    data = list(reader)

In [4]:
data

[['satisfaction_level',
  'last_evaluation',
  'number_project',
  'average_montly_hours',
  'time_spend_company',
  'Work_accident',
  'left',
  'promotion_last_5years',
  'sales',
  'salary'],
 ['0.38', '0.53', '2', '157', '3', '0', '1', '0', 'sales', 'low'],
 ['0.8', '0.86', '5', '262', '6', '0', '1', '0', 'sales', 'medium'],
 ['0.11', '0.88', '7', '272', '4', '0', '1', '0', 'sales', 'medium'],
 ['0.72', '0.87', '5', '223', '5', '0', '1', '0', 'sales', 'low'],
 ['0.37', '0.52', '2', '159', '3', '0', '1', '0', 'sales', 'low'],
 ['0.41', '0.5', '2', '153', '3', '0', '1', '0', 'sales', 'low'],
 ['0.1', '0.77', '6', '247', '4', '0', '1', '0', 'sales', 'low'],
 ['0.92', '0.85', '5', '259', '5', '0', '1', '0', 'sales', 'low'],
 ['0.89', '1', '5', '224', '5', '0', '1', '0', 'sales', 'low'],
 ['0.42', '0.53', '2', '142', '3', '0', '1', '0', 'sales', 'low'],
 ['0.45', '0.54', '2', '135', '3', '0', '1', '0', 'sales', 'low'],
 ['0.11', '0.81', '6', '305', '4', '0', '1', '0', 'sales', 'low'],
 

In [5]:
len(data)

15000

In [10]:
# Creación de la tabla donde se importará el archivo CSV
cur.execute("""
    CREATE TABLE Detalle (
        ID INTEGER PRIMARY KEY,
        satisfaction_level REAL,
        last_evaluation REAL,
        number_project INTEGER,
        average_montly_hours INTEGER,
        time_spend_company INTEGER,
        Work_accident INTEGER,
        left INTEGER,
        promotion_last_5years INTEGER,
        sales TEXT,
        salary TEXT
        );
""")

In [12]:
for renglon in range(1, len(data)):
    cur.execute("""
        INSERT INTO Detalle (satisfaction_level, last_evaluation, number_project, average_montly_hours, time_spend_company, Work_accident, left, promotion_last_5years, sales, salary)
        VALUES (?,?,?,?,?,?,?,?,?,?)
    """, data[renglon])

In [13]:
conn.commit()

In [15]:
# Satisfaccion promedio de empleados que abandonaron la empresa
cur.execute("SELECT AVG(satisfaction_level) FROM Detalle WHERE left = 1")
result = cur.fetchall()
result

[(0.44009801176141133,)]

In [16]:
# Satisfaccion promedio de empleados que no abandonaron la empresa
cur.execute("SELECT AVG(satisfaction_level) FROM Detalle WHERE left = 0")
result = cur.fetchall()
result

[(0.666809590479524,)]

Los empleados que deciden quedarse tienen un mayor nivel de satisfaccion promedio (66.68%)

In [18]:
cur.execute("SELECT AVG(average_montly_hours) FROM Detalle WHERE salary = 'low' OR salary = 'medium' GROUP BY salary")
result = cur.fetchall()
result

[(200.9965828321487,), (201.33834936394663,)]

Los empleados con salario LOW trabajaron en promedio 200.99 horas mensuales y los empleados con salario Medium trabajaron en promedio 201.33 horas mensuales

In [19]:
cur.execute("SELECT * FROM Detalle WHERE promotion_last_5years = 1 AND left = 1")
result = cur.fetchall()
result

[(19, 0.45, 0.51, 2, 160, 3, 1, 1, 1, 'sales', 'low'),
 (83, 0.79, 0.59, 4, 139, 3, 0, 1, 1, 'management', 'low'),
 (172, 0.41, 0.46, 2, 160, 3, 0, 1, 1, 'sales', 'low'),
 (288, 0.11, 0.79, 6, 292, 4, 0, 1, 1, 'technical', 'low'),
 (585, 0.41, 0.56, 2, 154, 3, 0, 1, 1, 'support', 'medium'),
 (752, 0.46, 0.45, 2, 138, 3, 0, 1, 1, 'IT', 'low'),
 (1001, 0.87, 1.0, 4, 258, 5, 1, 1, 1, 'sales', 'medium'),
 (1668, 0.44, 0.55, 2, 128, 3, 0, 1, 1, 'IT', 'medium'),
 (12019, 0.45, 0.51, 2, 160, 3, 1, 1, 1, 'sales', 'low'),
 (12083, 0.79, 0.59, 4, 139, 3, 0, 1, 1, 'management', 'low'),
 (12172, 0.41, 0.46, 2, 160, 3, 0, 1, 1, 'sales', 'low'),
 (12288, 0.11, 0.79, 6, 292, 4, 0, 1, 1, 'technical', 'low'),
 (12585, 0.41, 0.56, 2, 154, 3, 0, 1, 1, 'support', 'medium'),
 (12752, 0.46, 0.45, 2, 138, 3, 0, 1, 1, 'IT', 'low'),
 (14230, 0.45, 0.51, 2, 160, 3, 1, 1, 1, 'sales', 'low'),
 (14294, 0.79, 0.59, 4, 139, 3, 0, 1, 1, 'management', 'low'),
 (14383, 0.41, 0.46, 2, 160, 3, 0, 1, 1, 'sales', 'low'),
 

In [22]:
cur.execute("SELECT * FROM Detalle WHERE last_evaluation >= 0.9")
result = cur.fetchall()
result

[(9, 0.89, 1.0, 5, 224, 5, 0, 1, 0, 'sales', 'low'),
 (13, 0.84, 0.92, 4, 234, 5, 0, 1, 0, 'sales', 'low'),
 (18, 0.78, 0.99, 4, 255, 6, 0, 1, 0, 'sales', 'low'),
 (23, 0.09, 0.95, 6, 304, 4, 0, 1, 0, 'sales', 'low'),
 (26, 0.89, 0.92, 5, 242, 5, 0, 1, 0, 'sales', 'low'),
 (36, 0.1, 0.94, 6, 255, 4, 0, 1, 0, 'technical', 'low'),
 (55, 0.1, 0.92, 7, 307, 4, 0, 1, 0, 'support', 'low'),
 (57, 0.11, 0.94, 7, 255, 4, 0, 1, 0, 'support', 'low'),
 (60, 0.85, 1.0, 4, 225, 5, 0, 1, 0, 'technical', 'low'),
 (61, 0.85, 0.91, 5, 226, 5, 0, 1, 0, 'management', 'medium'),
 (62, 0.11, 0.93, 7, 308, 4, 0, 1, 0, 'IT', 'medium'),
 (63, 0.1, 0.95, 6, 244, 5, 0, 1, 0, 'IT', 'medium'),
 (65, 0.11, 0.94, 6, 286, 4, 0, 1, 0, 'IT', 'medium'),
 (68, 0.9, 0.98, 4, 264, 6, 0, 1, 0, 'product_mng', 'medium'),
 (71, 0.74, 0.99, 2, 277, 3, 0, 1, 0, 'IT', 'medium'),
 (75, 0.11, 0.97, 6, 277, 4, 0, 1, 0, 'product_mng', 'medium'),
 (92, 0.89, 1.0, 5, 246, 5, 0, 1, 0, 'sales', 'low'),
 (99, 0.11, 0.97, 6, 284, 4, 0, 1, 

In [23]:
# Cierre de la conexión
conn.close()